lang/src/org/partiql/lang/eval/builtins/UnixTimestampFunction.kt

```diff
@@ -0,0 +1,50 @@
+package org.partiql.lang.eval.builtins
+
+import org.partiql.lang.eval.Environment
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.lang.eval.NullPropagatingExprFunction
+import org.partiql.lang.eval.timestampValue
+import java.math.BigDecimal
+
+/**
+ * Builtin function to convert the given PartiQL `TIMESTAMP` [ExprValue] into a unix epoch, where a unix epoch
+ * represents the seconds since '1970-01-01 00:00:00' UTC. Largely based off MySQL's UNIX_TIMESTAMP.
+ *
+ * Syntax: `UNIX_TIMESTAMP([timestamp])`
+ *
+ * If UNIX_TIMESTAMP() is called with no [timestamp] argument, it returns the number of whole seconds since
+ * '1970-01-01 00:00:00' UTC as a PartiQL `INT` [ExprValue]
+ *
+ * If UNIX_TIMESTAMP() is called with a [timestamp] argument, it returns the number of seconds from
+ * '1970-01-01 00:00:00' UTC to the given [timestamp] argument. The return value will be a decimal if and only if
+ * the given [timestamp] has a fractional seconds part.
+ *
+ * The valid range of argument values is from '1970-01-01 00:00:01.000000' UTC up to the maximum PartiQL `TIMESTAMP`
+ * value. If given a [timestamp] out of this range, UNIX_TIMESTAMP() will return 0.
```

Comment:
Should we retun `0` here or `NULL`?

Reply:
I was following MySQL's version to convert timestamps -> unix epoch ([UNIX_TIMESTAMP](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_unix-timestamp)) which:
> ... If you pass an out-of-range date to UNIX_TIMESTAMP(), it returns 0.

Perhaps it would be better to keep it consistent with the other function to convert unix epoch -> timestamp. Currently that function will return a `NULL` `ExprValue` when given a unix epoch out of range.

Reply:
That is my dilemma as well. Personally I like the symmetry between the 2 functions and when the passed inputs result in an epoch that does not make sense, `NULL` seems more appropriate to me than `0`. Being "consistent" with other DBs will be less of a surprise to users. 

The alternative is to allow for "negative" values that denote time *before* 1970 which some epoch-functions tend to do (I think PostgreSQL does that).

Reply:
I like the alternative you suggested for allowing "negative" values to denote time _before_ 1970. Would be a pretty simple change to both functions.